flink-connectors/flink-connector-aws-kinesis-data-streams/src/test/java/org/apache/flink/connector/kinesis/table/KinesisDynamicTableSinkFactoryTest.java

```diff
@@ -0,0 +1,301 @@
+                                .setStream(STREAM_NAME)
+                                .setKinesisClientProperties(defaultProducerProperties())
+                                .setEncodingFormat(new TestFormatFactory.EncodingFormatMock(","))
+                                .setPartitioner(new RandomKinesisKeyGenerator<>())
+                                .build();
+
+        // verify that the constructed DynamicTableSink is as expected
+        Assertions.assertThat(actualSink).isEqualTo(expectedSink);
+
+        // verify the produced sink
+        DynamicTableSink.SinkRuntimeProvider sinkFunctionProvider =
+                actualSink.getSinkRuntimeProvider(new SinkRuntimeProviderContext(false));
+        Sink<RowData, ?, ?, ?> sinkFunction = ((SinkProvider) sinkFunctionProvider).createSink();
+        Assertions.assertThat(sinkFunction).isInstanceOf(KinesisDataStreamsSink.class);
+    }
+
+    @Test
+    public void testBadTableSinkForCustomPartitionerForPartitionedTable() {
+        ResolvedSchema sinkSchema = defaultSinkSchema();
+        Map<String, String> sinkOptions =
+                defaultTableOptions()
+                        .withTableOption(KinesisConnectorOptions.SINK_PARTITIONER, "random")
+                        .build();
+
+        Assertions.assertThatExceptionOfType(ValidationException.class)
+                .isThrownBy(
+                        () ->
+                                createTableSink(
+                                        sinkSchema, Arrays.asList("name", "curr_id"), sinkOptions))
+                .havingCause()
+                .withMessageContaining(
+                        String.format(
+                                "Cannot set %s option for a table defined with a PARTITIONED BY clause",
+                                KinesisConnectorOptions.SINK_PARTITIONER.key()));
+    }
+
+    @Test
+    public void testBadTableSinkForNonExistingPartitionerClass() {
+        ResolvedSchema sinkSchema = defaultSinkSchema();
+        Map<String, String> sinkOptions =
+                defaultTableOptions()
+                        .withTableOption(KinesisConnectorOptions.SINK_PARTITIONER, "abc")
+                        .build();
+
+        Assertions.assertThatExceptionOfType(ValidationException.class)
+                .isThrownBy(() -> createTableSink(sinkSchema, sinkOptions))
+                .havingCause()
+                .withMessageContaining("Could not find and instantiate partitioner class 'abc'");
+    }
+
+    private ResolvedSchema defaultSinkSchema() {
+        return ResolvedSchema.of(
+                Column.physical("name", DataTypes.STRING()),
+                Column.physical("curr_id", DataTypes.BIGINT()),
+                Column.physical("time", DataTypes.TIMESTAMP(3)));
+    }
+
+    private TableOptionsBuilder defaultTableOptionsWithSinkOptions() {
+        return defaultTableOptions()
+                .withTableOption("sink.fail-on-error", "true")
```

Comment:
Do we have constants for these? If not, should we? Might benefit people using Table API